package/src/hydra-ai/hydra-ai-backend.ts

```diff
@@ -1,23 +1,52 @@
 import "server-only";
 import AIService from "./ai-service";
 import { ComponentChoice } from "./model/component-choice";
-import { ComponentMetadata } from "./model/component-metadata";
+import { AvailableComponents, ComponentMetadata } from "./model/component-metadata";
 import { InputContext } from "./model/input-context";
+import { drizzle, NodePgDatabase } from "drizzle-orm/node-postgres";
+import { Pool } from "pg";
+import { registeredComponents } from "../db/schema";
+import { ComponentPropsMetadata } from "./model/component-props-metadata";
 
 export default class HydraBackend {
   private aiService: AIService;
+  private dbConnection: NodePgDatabase;
```

Comment:
Could be cool if we eventually use a generic "ComponentRepositoryInterface" so people can use whatever storage they want, but we can mess with that later.

Reply:
atm any postgres db should work.

what other storage methods were u thinking?

Reply:
For us I think postgres is good but for somebody who wants to self-host hydra it would be cool if they could use whatever they want, as long as they implement some interface

Reply:
i think postgres is most common for a lot of devs and we can add other implementations as the need arises.

if you've seen ppl request other data stores, i'm happy to add them?

i just think we should get the demo app built out first and can make the pkg more extensible later.

Reply:
yep agreed! No need to worry about it right now